test/integration/physics_system.cc

```diff
@@ -812,6 +817,387 @@ TEST_F(PhysicsSystemFixture, ResetVelocityComponent)
   EXPECT_NEAR(vel0, velocities[1], 0.05);
 }
 
+/////////////////////////////////////////////////
+/// Test joint position limit command component
+TEST_F(PhysicsSystemFixture, JointPositionLimitsCommandComponent)
+{
+  ignition::gazebo::ServerConfig serverConfig;
+
+  const auto sdfFile = std::string(PROJECT_SOURCE_PATH) +
+    "/test/worlds/revolute_joint.sdf";
+
+  sdf::Root root;
+  root.Load(sdfFile);
+  const sdf::World *world = root.WorldByIndex(0);
+  ASSERT_TRUE(nullptr != world);
+
+  serverConfig.SetSdfFile(sdfFile);
+
+  gazebo::Server server(serverConfig);
+
+  server.SetUpdatePeriod(1ms);
+
+  const std::string rotatingJointName{"j2"};
+
+  test::Relay testSystem;
+
+  // cppcheck-suppress variableScope
+  size_t iteration = 0u;
+
+  // The system is not in equilibrium at the beginning, so normally, joint j2
+  // would move. For the first 50 ms, we set position limits to 1e-6 so that
+  // it can't freely move, and we check it did not. For the other 50 ms, we
+  // remove the position limit and check that the joint has moved at least by
+  // 1e-2. Between times 30 and 40 ms we also add a 100 N force to the joint to
+  // check that the limit is held even in presence of force commands. Between
+  // times 40 and 50 ms, we add a velocity command to check that velocity
+  // commands do not break the positional limit.
+
+  testSystem.OnPreUpdate(
+    [&](const gazebo::UpdateInfo &, gazebo::EntityComponentManager &_ecm)
+    {
+      _ecm.Each<components::Joint, components::Name>(
+        [&](const ignition::gazebo::Entity &_entity,
+            const components::Joint *, components::Name *_name) -> bool
+        {
+          if (_name->Data() == rotatingJointName)
+          {
+            auto limitComp =
+                _ecm.Component<components::JointPositionLimitsCmd>(_entity);
+
+            if (iteration == 0u)
+            {
+              EXPECT_EQ(nullptr, limitComp);
+              _ecm.CreateComponent(_entity,
+                components::JointPositionLimitsCmd ({{-1e-6, 1e-6}}));
+              _ecm.CreateComponent(_entity, components::JointPosition());
+            }
+            else if (iteration == 50u)
+            {
+              EXPECT_NE(nullptr, limitComp);
+              limitComp->Data() = {{-1e6, 1e6}};
+            }
+            else
+            {
+              EXPECT_NE(nullptr, limitComp);
+              if (limitComp)
+              {
+                EXPECT_EQ(0u, limitComp->Data().size());
+              }
+              if (iteration >= 30u && iteration < 40u)
+              {
+                if (iteration == 30u)
```

Comment:
nit: Can this be simplified by using `_ecm.SetComponentData`?

Reply:
good idea, implemented in all relevant places